src/assets/AssetsDetectionController.ts

```diff
@@ -332,40 +332,47 @@ export class AssetsDetectionController extends BaseController<
         tokensToDetect.push(address);
       }
     }
+    const sliceOfTokensToDetect = [];
+    sliceOfTokensToDetect[0] = tokensToDetect.slice(0, 1000);
+    sliceOfTokensToDetect[1] = tokensToDetect.slice(
+      1000,
+      tokensToDetect.length - 1,
+    );
 
     const { selectedAddress } = this.config;
     /* istanbul ignore else */
     if (!selectedAddress) {
       return;
     }
-
-    await safelyExecute(async () => {
-      const balances = await this.getBalancesInSingleCall(
-        selectedAddress,
-        tokensToDetect,
-      );
-      const tokensToAdd = [];
-      for (const tokenAddress in balances) {
-        let ignored;
-        /* istanbul ignore else */
-        const { ignoredTokens } = this.getTokensState();
-        if (ignoredTokens.length) {
-          ignored = ignoredTokens.find(
-            (token) => token.address === toChecksumHexAddress(tokenAddress),
-          );
+    for (const tokensSlice of sliceOfTokensToDetect) {
+      await safelyExecute(async () => {
+        const balances = await this.getBalancesInSingleCall(
+          selectedAddress,
+          tokensSlice,
```

Comment:
This array might be empty if the user has less than 1000 tokens. Should we check whether it's empty or not before this point, and bail early if so?